packages/connection-storage/src/connection-storage.ts

```diff
@@ -1,520 +1,170 @@
-      .filter((x) => !x.favorite && x.id !== savedConnection.id)
-      .sort((a, b) => {
-        const aTime = a.lastUsed?.getTime() ?? 0;
-        const bTime = b.lastUsed?.getTime() ?? 0;
-        return bTime - aTime;
-      });
-
-    if (recentConnections.length >= this.maxAllowedRecentConnections) {
-      await this.delete({
-        id: recentConnections[recentConnections.length - 1].id,
-      });
-    }
+export class NoopCompassConnectionStorage
+  extends NoopConnectionStorage
+  implements CompassConnectionStorage
+{
+  constructor() {
+    super();
   }
 
-  static async deserializeConnections({
-    content,
-    options = {},
-    signal,
-  }: {
-    content: string;
-    options: ImportConnectionOptions;
-    signal?: AbortSignal;
-  }): Promise<ConnectionInfo[]> {
-    throwIfAborted(signal);
-
-    const { passphrase, trackingProps, filterConnectionIds } = options;
-    const connections = await deserializeConnections(content, {
-      passphrase,
-      trackingProps,
-    });
-    return filterConnectionIds
-      ? connections.filter((x) => filterConnectionIds.includes(x.id))
-      : connections;
+  getLegacyConnections(): Promise<{ name: string }[]> {
+    return Promise.resolve([]);
   }
-
-  static async importConnections({
-    content,
-    options = {},
-    signal,
-  }: {
-    content: string;
-    options?: ImportConnectionOptions;
-    signal?: AbortSignal;
-  }): Promise<void> {
-    throwIfAborted(signal);
-
-    const connections = await this.deserializeConnections({
-      content,
-      options,
-      signal,
-    });
-
-    log.info(
-      mongoLogId(1_001_000_149),
-      'Connection Import',
-      'Starting connection import',
-      {
-        count: connections.length,
-      }
-    );
-
-    await Promise.all(
-      connections.map((connectionInfo) => this.save({ connectionInfo, signal }))
-    );
-
-    log.info(
-      mongoLogId(1_001_000_150),
-      'Connection Import',
-      'Connection import complete',
-      {
-        count: connections.length,
-      }
-    );
+  deserializeConnections(): Promise<ConnectionInfo[]> {
+    return Promise.resolve([]);
   }
-
-  static async exportConnections({
-    options = {},
-    signal,
-  }: {
-    options?: ExportConnectionOptions;
-    signal?: AbortSignal;
-  } = {}): Promise<string> {
-    throwIfAborted(signal);
-
-    const { filterConnectionIds, ...restOfOptions } = options;
-    const connections = await this.loadAll({ signal });
-    const exportConnections = filterConnectionIds
-      ? connections.filter((x) => filterConnectionIds.includes(x.id))
-      : connections.filter((x) => x.favorite?.name);
-
-    return serializeConnections(exportConnections, restOfOptions);
+  exportConnections(): Promise<string> {
+    return Promise.resolve('');
+  }
+  importConnections(): Promise<void> {
+    return Promise.resolve();
   }
 }
+
+export function isCompassConnectionStorage(
```

Comment:
See my other comment related to that, I'd say we probably want to avoid having two completely separate interfaces here